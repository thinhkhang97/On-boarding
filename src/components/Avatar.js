import * as React from 'react';
import { Image, StyleSheet } from 'react-native';

type Props = {
  source: string,
  style?: object
};

class Avatar extends React.Component<Props> {
  static defaultObject = {
    style: {
      height: 116,
      width: 116,
      borderRadius: 116 / 2
    }
  };
  render() {
    return <Image source={this.props.source} style={this.props.style} />;
  }
}

export default Avatar;
