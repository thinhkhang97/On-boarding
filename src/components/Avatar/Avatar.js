// @flow

import * as React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import type { ____ViewStyleProp_Internal as Style } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

type Props = {
  source: {
    uri: string,
  },
  style: Style,
  size: number,
};

class Avatar extends React.Component<Props> {
  static defaultProps = {
    size: 116,
    style: {},
  };
  render() {
    const { size, source, style } = this.props;
    return (
      <Image
        source={source}
        style={[
          {
            height: size,
            width: size,
            borderRadius: size / 2,
          },
          style,
        ]}
      />
    );
  }
}

export default Avatar;
