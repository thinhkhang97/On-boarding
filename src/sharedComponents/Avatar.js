// @flow
import * as React from 'react';
import { Image, StyleSheet } from 'react-native';

type Props = {
  source: string,
  style?: View.propTypes.style,
  size?: number,
};

class Avatar extends React.Component<Props> {
  static defaultProps = {
    size: 116,
    style: {},
  };
  render() {
    const { size, source, style } = this.props;
    return (
      <Image
        source={source}
        style={[
          {
            height: size,
            width: size,
            borderRadius: size / 2,
          },
          style,
        ]}
      />
    );
  }
}

export default Avatar;
