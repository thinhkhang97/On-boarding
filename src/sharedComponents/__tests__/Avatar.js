import * as React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../Avatar';

test('Render avatar correctly', ()=>{
    const testAvatar = renderer.create(<Avatar/>).toJSON();
    expect(testAvatar).toMatchSnapshot();
})