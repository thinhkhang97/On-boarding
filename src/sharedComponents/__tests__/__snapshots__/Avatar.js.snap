// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avatar correctly 1`] = `
<Image
  style={
    Array [
      Object {
        "borderRadius": 58,
        "height": 116,
        "width": 116,
      },
      Object {},
    ]
  }
/>
`;
